pkg/api/v1/pod/util.go

```diff
@@ -63,21 +63,30 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 
 // TODO: remove this function when init containers becomes a stable feature
 func SetInitContainersAndStatuses(pod *v1.Pod) error {
-	var initContainersAnnotation string
-	initContainersAnnotation = pod.Annotations[v1.PodInitContainersAnnotationKey]
-	initContainersAnnotation = pod.Annotations[v1.PodInitContainersBetaAnnotationKey]
-	if len(initContainersAnnotation) > 0 {
+	// If there is a beta annotation, copy to alpha key.
+	// See commit log for PR #31026 for why we do this.
+	if valueBeta, okBeta := pod.Annotations[v1.PodInitContainersBetaAnnotationKey]; okBeta {
+		pod.Annotations[v1.PodInitContainersAnnotationKey] = valueBeta
```

Comment:
API servers are required to be updated before nodes, so 

1. node owns which field it looks at first (old versions only look at beta, new versions look at beta, then field)
2. when a kubelet updates, it definitely needs to look at beta status annotation, but it can ignore beta spec annotation (because api server is required to be newer)